### PR TITLE
Log console output during replay to file #985

### PR DESCRIPTION
**Pull request by cogutvalera**
PR for #985 **"Log console output during replay to file"**

bump fc and changed (std::cerr) to (ilog) in application.cpp and witness.cpp

### CODE REVIEW

File: libraries/app/application.cpp [hunk 1]
```diff
@@ -337,20 +337,23 @@ void application_impl::startup()
...
             FC_ASSERT( genesis.initial_witness_candidates.size() >= genesis.initial_active_witnesses );
             set_dbg_init_key( genesis, init_key );
             modified_genesis = true;
-            std::cerr << "Set init witness key to " << init_key << "\n";
+            ilog("Set init witness key to ${init_key}\n", ("init_key", init_key));
          }
          if( modified_genesis )
          {
-            std::cerr << "WARNING:  GENESIS WAS MODIFIED, YOUR CHAIN ID MAY BE DIFFERENT\n";
+            ilog("WARNING:  GENESIS WAS MODIFIED, YOUR CHAIN ID MAY BE DIFFERENT\n");
```
Comment (pmconrad) [inline libraries/app/application.cpp]: Please use wlog here

File: libraries/app/application.cpp [hunk 2]
```diff
@@ -982,16 +985,18 @@ void application::initialize(const fc::path& data_dir, const boost::program_opti
          try {
             genesis_state = fc::json::from_file(genesis_out).as<genesis_state_type>( 20 );
          } catch(const fc::exception& e) {
-            std::cerr << "Unable to parse existing genesis file:\n" << e.to_string()
-                      << "\nWould you like to replace it? [y/N] ";
+            ilog(
```
Comment (pmconrad) [inline libraries/app/application.cpp]: This prompts for user input, please change back to original.

File: libraries/plugins/witness/witness.cpp [hunk 3]
```diff
@@ -41,21 +41,21 @@ namespace bpo = boost::program_options;
...
       "*   - Welcome to Graphene! -   *\n"
       "*   ------------------------   *\n"
       "*                              *\n"
       "********************************\n"
-      "\n";
+      "\n");
    if( db.get_slot_at_time( fc::time_point::now() ) > 200 )
    {
-      std::cerr << "Your genesis seems to have an old timestamp\n"
+      ilog("Your genesis seems to have an old timestamp\n"
```
Comment (pmconrad) [inline libraries/plugins/witness/witness.cpp]: Please change to wlog.

File: libraries/app/application.cpp (hunk 1 above)
Comment (cogutvalera) [reply libraries/app/application.cpp]: ok sure ! Thanks !

File: libraries/app/application.cpp (hunk 2 above)
Comment (cogutvalera) [reply libraries/app/application.cpp]: ok sure ! Thanks !

File: libraries/plugins/witness/witness.cpp (hunk 3 above)
Comment (cogutvalera) [reply libraries/plugins/witness/witness.cpp]: ok sure ! Thanks !

File: libraries/app/application.cpp [hunk 4]
```diff
@@ -337,20 +337,23 @@ void application_impl::startup()
...
+            );
          }
          if( _options->count("dbg-init-key") )
          {
             std::string init_key = _options->at( "dbg-init-key" ).as<string>();
             FC_ASSERT( genesis.initial_witness_candidates.size() >= genesis.initial_active_witnesses );
             set_dbg_init_key( genesis, init_key );
             modified_genesis = true;
-            std::cerr << "Set init witness key to " << init_key << "\n";
+            ilog("Set init witness key to ${init_key}\n", ("init_key", init_key));
```
Comment (abitmore) [inline libraries/app/application.cpp]: `\n` is not needed in `ilog`.

File: libraries/app/application.cpp [hunk 5]
```diff
@@ -337,20 +337,23 @@ void application_impl::startup()
             genesis.initial_timestamp -= ( genesis.initial_timestamp.sec_since_epoch()
                                            % genesis.initial_parameters.block_interval );
             modified_genesis = true;
-            std::cerr << "Used genesis timestamp:  " << genesis.initial_timestamp.to_iso_string()
-                      << " (PLEASE RECORD THIS)\n";
+
+            ilog(
+               "Used genesis timestamp:  ${timestamp} (PLEASE RECORD THIS)\n", 
+               ("timestamp", genesis.initial_timestamp.to_iso_string())
+            );
```
Comment (abitmore) [inline libraries/app/application.cpp]: `\n` is not needed in `ilog`.

File: libraries/app/application.cpp [hunk 6]
```diff
@@ -989,9 +992,9 @@ void application::initialize(const fc::path& data_dir, const boost::program_opti
                return;
          }
 
-         std::cerr << "Updating genesis state in file " << genesis_out.generic_string() << "\n";
+         ilog("Updating genesis state in file ${genesis_out} \n", ("genesis_out", genesis_out.generic_string()));
```
Comment (abitmore) [inline libraries/app/application.cpp]: I don't think here and below should be changed to `ilog`. @pmconrad what's your opinion?

File: libraries/plugins/witness/witness.cpp [hunk 7]
```diff
@@ -41,21 +41,21 @@ namespace bpo = boost::program_options;
...
       "*   - Welcome to Graphene! -   *\n"
       "*   ------------------------   *\n"
       "*                              *\n"
       "********************************\n"
-      "\n";
+      "\n");
    if( db.get_slot_at_time( fc::time_point::now() ) > 200 )
    {
-      std::cerr << "Your genesis seems to have an old timestamp\n"
+      wlog("Your genesis seems to have an old timestamp\n"
```
Comment (abitmore) [inline libraries/plugins/witness/witness.cpp]: Personally I think having `\n` in the middle of a log message is a bit strange. What's your opinion?

File: libraries/app/application.cpp [hunk 8]
```diff
@@ -337,20 +337,23 @@ void application_impl::startup()
...
             FC_ASSERT( genesis.initial_witness_candidates.size() >= genesis.initial_active_witnesses );
             set_dbg_init_key( genesis, init_key );
             modified_genesis = true;
-            std::cerr << "Set init witness key to " << init_key << "\n";
+            ilog("Set init witness key to ${init_key}\n", ("init_key", init_key));
          }
          if( modified_genesis )
          {
-            std::cerr << "WARNING:  GENESIS WAS MODIFIED, YOUR CHAIN ID MAY BE DIFFERENT\n";
+            wlog("WARNING:  GENESIS WAS MODIFIED, YOUR CHAIN ID MAY BE DIFFERENT\n");
```
Comment (abitmore) [inline libraries/app/application.cpp]: `\n` is not needed.

**Comment by cogutvalera** [comment]
Thanks !

**Pull request merged**